### PR TITLE
Add `typos` + `cargo doc` CI

### PR DESCRIPTION
**Pull request by hinto-janai**
- Adds https://github.com/crate-ci/typos to CI
- Adds `cargo doc` checks to CI
- Adds `typos.toml` for false-positives and excluded files
- Fixes existing typos
- Changes `cargo fmt` to only run on 1 OS

### CODE REVIEW

**Comment by hinto-janai** [comment]
@Boog900 ready.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -84,19 +96,23 @@ jobs:
...
-      run: cargo clippy --workspace --all-targets --all-features -- -D warnings
+      run: cargo clippy --workspace --all-features -- -D warnings
 
     - name: Test
-      run: cargo test --all-features --workspace --all-targets
+      run: cargo test --all-features --workspace
 
     # TODO: upload binaries with `actions/upload-artifact@v3`
     - name: Build
-      run: cargo build --all-features --all-targets --workspace
```
Comment (Boog900) [inline .github/workflows/ci.yml:102]: `--all-targets` does not mean all target archs it means tests, benches, libs etc so we should keep it for `clippy` and `build`. For `test` it seems specifying  `--all-targets` acctually reduces the amount of our tests ran: it skips docs tests so we should keep `--all-targets` removed for that

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,16 +13,18 @@ env:
   RUST_BACKTRACE: "full"
   # Increase thread stack size to 8 megabytes.
   RUST_MIN_STACK: 8000000
+  # Fail on documentation warnings.
+  RUSTDOCFLAGS: '-D warnings'
```
Comment (Boog900) [inline .github/workflows/ci.yml:17]: can this be passed in like we do with clippy or can you not do that with `doc`?

**Review by Boog900 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hinto-janai) [reply .github/workflows/ci.yml:17]: You cannot - we could do `RUSTDOCFLAGS='-D warnings' cargo doc` but that causes a force-rebuild, so it's set for the whole CI environment.

**Comment by hinto-janai** [comment]
`handshake_cuprate_to_monerod` failed https://github.com/Cuprate/cuprate/actions/runs/7909525054/job/21590698059?pr=32#step:11:490.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Boog900) [reply .github/workflows/ci.yml:17]: ah alright yeah this is better then

**Review by Boog900 — APPROVED** [review]
It passed after a re-run hopefully that doesn't happen too often.

**Pull request merged**